File: src/calisma2/main.java
package calisma2;

import java.util.Scanner;

public class main {
    public static void main(String[] args) {

       /*
       Yazılan bir keilmeyi önce 16 lık tabana,sonra 2 lik tabana dönüştürme.
       ilk olarak bunların  ASCII değerini bulmamız lazım. ASCII değeri,bilgisyarın bu harf
       karakterlerini işlemesi için gerekli ve her bir harfın bir ASCII değeri var.ve ascıı değeri decimaldir.
        */
        Scanner scan = new Scanner(System.in);
        System.out.println("Kelime gir: ");
        String kelime=scan.nextLine();

        // burada kelimenin her karakterinin ASCII değerini alıyoruz dönüştürüyoruz.
        String hexdeger= " ";
        for(char i : kelime.toCharArray())
        {
            int asciiDeger=(int) i;
            hexdeger=hexdeger+decimalToHex(asciiDeger);

        }
        System.out.println("kelimenin hexadecimal karşılığı: "+hexdeger);
        String binarydeger=hexToBinary(hexdeger);
        System.out.println("kelimenin binary karşılığı: "+binarydeger);
    }

    public static String decimalToHex(int dc){
        String hexdeger =" ";
        while(dc>0)
        {
            int kalan = dc %16;
            hexdeger= convertToHex(kalan ) +hexdeger;
            dc=dc/16;
        }
        while (hexdeger.length() < 2)
        {
            hexdeger = "0" + hexdeger;
        }
        return hexdeger.isEmpty() ? "00" : hexdeger;
    }
    public static char convertToHex(int a) {
        if (a >= 0 && a <= 9) { //burada 0-9 arasındakiler için önce askiiye ordan karakter karşılığına çeviriyoruz
            return (char) (a + '0');
        } else if (a >= 10 && a <= 15) { // 10-15 arası için
            return (char) (a - 10 + 'A');
        }
        return '?';  }
    public static String hexToBinary(String hexdeger){
        StringBuilder binaryString = new StringBuilder();

        for (char hexChar : hexdeger.toCharArray()) {
            binaryString.append(hexCharToBinary(hexChar));
        }

        return binaryString.toString();
    }
    public static String hexCharToBinary(char hexChar){
        // Hexadecimal karakterini integer değerine dönüştür
        int decimaldeger = hexChar >= '0' && hexChar <= '9' ? hexChar - '0' : (hexChar - 'A' + 10);

        // Decimal değeri 4-bit binary formata dönüştür
        String binaryValue = Integer.toBinaryString(decimaldeger);
        return String.format("%4s", binaryValue).replace(' ', '0');

    }
    }

File: src/calisma2/main.kt
package calisma2

fun main() {
    /*
       Yazılan bir kelimeyi önce 16'lık tabana, sonra 2'lik tabana dönüştürme.
       İlk önce bunların ASCII değerini bulmamız lazım. ASCII değeri,bilgisayarın harfleri
       kendine göre işleyebilmesi için bayağı gerekli.her bir harfin bir ASCII değeri var ve ASCII değeri decimaldir.
    */
    val kelime: String? = readLine()
    val asciidegerler = kelime?.map { it.toInt() } // kelimenin içindeki her karakterin ASCII değerini alır

    // Her bir ASCII değerini hexadecimal formatına döndürüyor  sonra,
    // bu değerleri de aralarına boşluk koyarak birleştiriyor
    val hexdeger = asciidegerler?.joinToString(separator = " ") { decimalToHex(it) }

    // Hexadecimal değerleri alıp, her birini binary formatına döndürür ve aralarına boşluk koyar
    val binarydeger = asciidegerler?.joinToString(separator = " ") { decimalToBinary(it) }

    println("Hexadecimal değeri: $hexdeger")
    println("Binary değeri: $binarydeger")
}

fun decimalToHex(decimal: Int): String {
    if (decimal == 0) return "0"
    var sayi = decimal
    val hexKarakterleri = "0123456789ABCDEF"
    var hexKelime = ""

    while (sayi > 0) {
        val kalan = sayi % 16
        hexKelime = hexKarakterleri[kalan] + hexKelime
        sayi /= 16
    }
    return hexKelime
}
fun decimalToBinary(decimal: Int): String {
    if (decimal == 0) return "0000"
    var sayi = decimal
    var binaryKelime = ""

    while (sayi > 0) {
        val kalan = sayi % 2
        binaryKelime = kalan.toString() + binaryKelime
        sayi /= 2
    }

    // binary kelimenin uzunluğunu 4 haneli yapıyo burada
    while (binaryKelime.length < 4) {
        binaryKelime = "0" + binaryKelime
    }

    return binaryKelime
}
